packages/medusa/src/api/routes/store/products/list-products.ts

```diff
@@ -64,17 +92,68 @@ export default async (req, res) => {
   })
 }
 
-export class StoreGetProductsParams {
-  @IsInt()
+export class StoreGetProductsPaginationParams {
+  @IsNumber()
+  @IsOptional()
   @Type(() => Number)
-  limit = 100
+  offset?: number = 0
 
-  @IsInt()
+  @IsNumber()
+  @IsOptional()
   @Type(() => Number)
-  offset = 0
+  limit?: number = 100
+}
+
+export class StoreGetProductsParams extends StoreGetProductsPaginationParams {
+  @IsString()
+  @IsOptional()
+  id?: string
+
+  @IsString()
+  @IsOptional()
+  q?: string
+
+  @IsArray()
+  @IsOptional()
+  collection_id?: string[]
+
+  @IsArray()
+  @IsOptional()
+  tags?: string[]
+
+  @IsString()
+  @IsOptional()
+  title?: string
 
+  @IsString()
   @IsOptional()
+  description?: string
+
+  @IsString()
+  @IsOptional()
+  handle?: string
+
   @IsBoolean()
+  @IsOptional()
   @Type(() => Boolean)
   is_giftcard?: boolean
```

Comment:
Should be changed to parse string `true` || `false` as boolean:
```suggestion
  @IsBooleanString()
  @IsOptional()
  is_giftcard?: boolean
```